runtime/src/bank.rs

```diff
@@ -890,6 +900,27 @@ impl Bank {
         new
     }
 
+    fn enable_stake_rewrite(&self, parent_epoch: Epoch) -> bool {
+        if self.no_stake_rewrite.load(Relaxed) {
+            return false;
+        }
+
+        let activated_slot = if let Some(epoch) = self.rewrite_stake_slot() {
+            epoch
+        } else {
+            return false;
+        };
+        if activated_slot == 0 {
+            return self.slot() == 0;
+        }
+        let (activated_epoch, _) = self.get_epoch_and_slot_index(activated_slot);
+
+        // to avoid any potential risk of wrongly rewriting accounts in the future,
+        // only do this once, taking small risk of unknown
+        // bugs which again creates bad stake accounts..
+        self.epoch() > parent_epoch && self.epoch() == activated_epoch
```

Comment:
I don't grok why we need to check epochs here. Wouldn't `self.slot() == activated_slot` be correct?

Reply:
well, there is possibility of skipped slots. yeah hard to read correctly. ;)

Reply:
How could the activation slot be skipped? It's set in `new.apply_feature_activations() as `self.slot()` immediately before: https://github.com/solana-labs/solana/blob/acc40b1c9e54c7277f5369d20cd9f1e2fc88d1e7/runtime/src/bank.rs#L4116

Reply:
yeah, i was wrong... fixed: https://github.com/solana-labs/solana/pull/13461/commits/4410a3599fa376e3eba82832c9222c8ab47798b3